### DISCUSSION TITLE
setting and using the project/working directory for rendering

### DISCUSSION THREAD

**Discussion by shaenzi** [quarto-dev/quarto-cli#706] [Q&A]
Hi,
I am currently struggling to run my code from the base project directory when I use `quarto render`. What I have done so far:
- Following [the project basics](https://quarto.org/docs/projects/quarto-projects.html), I have run `quarto create-project`in my already existing repo, which created a `_quarto.yml` file, consisting of a project title
- I have a .qmd file in a subfolder of the base project (subfolder is called notebooks), where I specify the execute-dir as project, as described [here](https://quarto.org/docs/projects/quarto-projects.html)
- then I render this document, with the code printing the working directory, which still is the subfolder, not the main folder of the repo. 

Did I miss anything in setting the project up?
Context info: running python, using VSCode, with a jupyter kernel.
thanks!

**Comment by shaenzi** [comment]
Additional info: the `project: execute-dir` does not seem to have any effect. Same behaviour with `project` and `file`, and strange historicity when the file is run several times: seems to remember the last location when there is some code run that changes the directory, rather than starting afresh.

**Comment by jjallaire** [comment]
Thanks for reporting this! Should be fixed here: https://github.com/quarto-dev/quarto-cli/commit/a540075c764d3ee1756041189e96c53ee42ebf94

**Reply by shaenzi** [reply]
Hi,
thanks for your quick reply and suggested fix! However, even after upgrading to the latest version (0.9.305, which should include your fix), the behaviour is still inconsistent on my side. I can set 
```
project:
  execute-dir: file
```
and land in the main folder if I happened to be there in the previous run from some code that changes directories, and I can be in the subfolder when I set it like this: 
```
project:
  execute-dir: project
```

The way I check in python where I am in a code cell is like this:
```
import os
print(os.getcwd())
```

Any ideas? suggestions of what I should try?
thanks!

**Reply by jjallaire** [reply]
I'm not able to reproduce this (the directory is correctly set and reset for new renders). Here's the test project I am using: https://github.com/jjallaire/execute-dir

When you updated it's possible that you were still running the old version when you tested. Could you make 100% sure you are running the latest version? If that doesn't work LMK what platform you are running on as it's possible (but not very likely) that there is a platform difference in play (I'm on OS X)

**Reply by shaenzi** [reply]
I'm on Windows. updated to latest version (0.9.342) and behaviour is still inconsistent. Will use a workaround for now and test on another windows machine later.

**Reply by jjallaire** [reply]
Interesting -- I still think you may not be running the right version as I tried on Windows and found that the revised implementation actually results in a runtime error (because windows paths with "\" are not properly escaped). So on Windows I'd expect that you would see this error rather than the no-op you seem to be using.

The error with the bashslash escaping is fixed here: https://github.com/quarto-dev/quarto-cli/commit/d64661dd68174cfd0fb9e869a9038acf30b1065e
